lib/backend/k8s/resources/profile.go

```diff
@@ -237,40 +270,44 @@ func (c *profileClient) Watch(ctx context.Context, list model.ListInterface, rev
 		"saRev": saRev,
 	}).Debug("Watching two resources at individual revisions")
 
-	nsWatch, err := c.clientSet.CoreV1().Namespaces().Watch(metav1.ListOptions{ResourceVersion: nsRev})
-	if err != nil {
-		return nil, K8sErrorToCalico(err, list)
+	opts.ResourceVersion = nsRev
+	var nsWatch kwatch.Interface = kwatch.NewFake()
+	if namePrefix == "" || namePrefix == conversion.NamespaceProfileNamePrefix {
+		nsWatch, err = c.clientSet.CoreV1().Namespaces().Watch(opts)
+		if err != nil {
+			return nil, K8sErrorToCalico(err, list)
+		}
 	}
 	converter := func(r Resource) (*model.KVPair, error) {
 		k8sNamespace, ok := r.(*kapiv1.Namespace)
 		if !ok {
-			return nil, errors.New("profile conversion with incorrect k8s resource type")
+			return nil, errors.New("Profile conversion with incorrect k8s resource type")
```

Comment:
In golang, errors typically always start with a lowercase so that they can be more easily chained. Probably OK here since we're referring to a specific type, but thought it worth calling out.

Reply:
I was trying to be consistent with the rest of the file. Specifically with [line 304](https://github.com/projectcalico/libcalico-go/blob/c9b134e0e76ac14882aefcd701d979043956bcb4/lib/backend/k8s/resources/profile.go#L304) with the same error message just a few lines below.